eclair-core/src/main/scala/fr/acinq/eclair/db/PendingRelayDb.scala

```diff
@@ -17,30 +17,28 @@
 package fr.acinq.eclair.db
 
 import fr.acinq.bitcoin.ByteVector32
-import fr.acinq.eclair.channel.Command
+import fr.acinq.eclair.channel.HasHtlcIdCommand
 
 /**
-  * This database stores the preimages that we have received from downstream
-  * (either directly via UpdateFulfillHtlc or by extracting the value from the
-  * blockchain).
-  *
-  * This means that this database is only used in the context of *relaying* payments.
-  *
-  * We need to be sure that if downstream is able to pulls funds from us, we can always
-  * do the same from upstream, otherwise we lose money. Hence the need for persistence
-  * to handle all corner cases.
-  *
-  */
+ * This database stores the preimages that we have received from downstream
+ * (either directly via UpdateFulfillHtlc or by extracting the value from the
+ * blockchain).
+ *
+ * This means that this database is only used in the context of *relaying* payments.
+ *
+ * We need to be sure that if downstream is able to pulls funds from us, we can always
+ * do the same from upstream, otherwise we lose money. Hence the need for persistence
+ * to handle all corner cases.
+ *
+ */
 trait PendingRelayDb {
 
-  def addPendingRelay(channelId: ByteVector32, htlcId: Long, cmd: Command)
+  def addPendingRelay(channelId: ByteVector32, cmd: HasHtlcIdCommand)
 
   def removePendingRelay(channelId: ByteVector32, htlcId: Long)
 
-  def listPendingRelay(channelId: ByteVector32): Seq[Command]
-
-  def listPendingRelay(): Set[(ByteVector32, Long)]
+  def listPendingRelay(channelId: ByteVector32): Seq[HasHtlcIdCommand]
 
-  def close(): Unit
+  def listPendingRelay(): Set[(ByteVector32, HasHtlcIdCommand)]
```

Comment:
Why do you change the signature? We currently only need `(ByteVector32, Long)` here which is more efficient to retrieve from the DB that decoding the command (especially since we don't use it).

Reply:
Returning `HasHtlcIdCommand` seems more general and can be used for other purposes in the future, also currently `listPendingRelay` is only used on startup once so efficiency does not seem that important. This is not a big deal though, I can change it back.

Reply:
To me this is YAGNI (you ain't gonna need it) :)
I would change that only when we actually need the whole command somewhere, especially since as you pointed out in other discussions we're reading from `pendingRelay` quite often (channel state changes) so we'd like it to be as lightweight as possible (even though it's not very lightweight).